### ISSUE TITLE
Failing Conditional Update within Transaction: 'Bundle.Entry.resource must contain an id field for a PUT operation.'

### ISSUE THREAD

**Issue by chgl** [IBM/FHIR#1319]
**Describe the bug**
When sending a transaction containing a conditional update operation where `resource.id` is not specified, the operation fails with `Bundle.Entry.resource must contain an id field for a PUT operation.` despite this not being a requirement for conditional updates.

**To Reproduce**
1. Go to 'https://www.hl7.org/fhir/bundle-transaction.json.html'
2. Copy the transaction JSON
3. Send it to the FHIR Server via curl

The (simplified) failing entry in the sample transaction is:

```json
{
  "resourceType": "Bundle",
  "id": "bundle-transaction",
  "type": "transaction",
  "entry": [
    {
      "fullUrl": "urn:uuid:74891afc-ed52-42a2-bcd7-f13d9b60f096",
      "resource": {
        "resourceType": "Patient",
        "identifier": [
          {
            "system": "http:/example.org/fhir/ids",
            "value": "456456"
          }
        ],
        "birthDate": "1974-12-25"
      },
      "request": {
        "method": "PUT",
        "url": "Patient?identifier=http:/example.org/fhir/ids|456456"
      }
    }
  ]
}
```

**Expected behavior**
Instead of returning an error, the server should accept the transaction, check if a Patient resource with the given `identifier` (search expression) is found. If yes, it should update the contents of the resource with data provided in the transaction. If no resource with the given `identifier` is found, a new resource should be generated with the provided contents. Ie following the specification here: https://www.hl7.org/fhir/http.html#cond-update

> The resource MAY contain an id element, but does not need to (this is one of the few cases where a resource exists without an id element). If an id is provided, the server SHALL ignore it - see documentation for the update interaction.

**Additional context**

If a resource id is specified but changed between two `POST`s, while the search expression stays the same, it fails with `Input resource 'id' attribute must match the id of the search result resource.`

https://www.hl7.org/fhir/http.html#cond-update says

> Unlike this rest of this page, the conditional create, update, patch and delete are trial use until further experience is gained with their use. Their status will be reviewed in a future version of FHIR.

So I understand if this is something you are currently not tackling. A workaround for some cases might be to use update-as-create.

**Comment by prb112** [comment]
Fix is in 4.3.1

**Comment by chgl** [comment]
@prb112 awesome! Thanks for quick fix, @JohnTimm
